vote.py

```diff
@@ -0,0 +1,190 @@
+        if question:
+            return self.voting == self.VOTING_QUESTION
+        return self.voting is not self.VOTING_NONE
+
+    def judgement(self, event=None, penalty=''):
+        """Summarize votes and make decision"""
+        # TODO: if voting succeeded, get op and punish user
+        msg = 'unsuccessfully'
+        if self.yeses + self.nos > 1 and self.yeses > self.nos:
+            msg = 'successfully '
+
+        msg += ' with {0}x yes, {1}x no'.format(self.yeses, self.nos)
+        result = '{0} {1}'.format(penalty, event.args[0])
+
+        self.reset()
+
+        return 'Voting to {0} finished {1}'.format(result, msg)
+
+    def vote_result(self, event=None, penalty=''):
+        """Start voting of given penalty"""
+        if self.voting_in_progress() or self.voting_in_progress(question=True):
+            return 'Voting in progress...'
+
+        args = event.args
+        if not len(args):
+            return 'No user specified'
+
+        if '-' in args:
+            args = split_args_by(args, '-')
+
+        try:
+            time = int(args[1])
+        except:
+            time = 60
+
+        user = args[0]
+        self.voting = self.VOTING_YESNO
+
+        self.delay_task(time, self.judgement, event=event, penalty=penalty)
+
+        return 'Voting to {0} {1} started for {2} secs. (!yes / !no)' \
+            .format(penalty, user, time)
+
+    # TODO: put all these commands into one regexp match
+    @cmd
+    def votekick(self, event):
+        return self.vote_result(event, 'kick')
+
+    @cmd
+    def voteban(self, event):
+        return self.vote_result(event, 'ban')
+
+    @cmd
+    def voteunban(self, event):
+        return self.vote_result(event, 'unban')
+
+    @cmd
+    def votemute(self, event):
+        return self.vote_result(event, 'mute')
+
+    def yesno_result(self, event=None):
+        """Return results of yes/no question"""
+        yes = []
+        no = []
+        for k, v in self.voted.iteritems():
+            if v[1]:
+                yes.append(v[0])
+            elif not v[1]:
+                no.append(v[0])
+
+        y = ' ({0})'.format(', '.join(yes)) if len(yes) else ''
+        n = ' ({0})'.format(', '.join(no)) if len(no) else ''
+        res = '{0}x yes{2}, {1}x no{3}'.format(self.yeses, self.nos, y, n)
+
+        self.reset()
+        return 'Results of {0}\'s question: {1}'.format(event.meta['nick'],
+                                                        res)
+
+    def question_helper(self, event, callback, yesno=False):
+        """Helper which which holds `ask` commands"""
+        if self.voting_in_progress() or self.voting_in_progress(question=True):
+            return 'Voting in progress...'
+
+        args = split_args_by(event.args, '-')
+        if not len(args):
+            return 'No question specified'
+
+        question = args[0]
+        if not question.endswith('?'):
+            question += '?'
+
+        time = 60 if len(args) == 1 else int(args[1])
```

Comment:
@Adman so if `args[1]` won't be an int, this will just throw an exception that is not caught anywhere, right?